lib/authOptions.ts

```diff
@@ -59,27 +40,26 @@ export const authOptions: NextAuthOptions = {
     }),
     Credentials({
       name: "Credentials",
-      // TODO: im a little confused as to how the signin flow works, since a signUp needs to create a new user and the form has
-      // TODO: extra data (full name) and needs some way of flagging this afterr sign up to redirect to onboarding
+
       credentials: {
         username: { label: "Email", type: "email", placeholder: "..." },
         password: { label: "Password", type: "password" },
       },
 
       async authorize(credentials, req) {
-        // TODO: confused about where exactly things need to be encapsulated regarding server/client side logic asnd this file and the button
-        // that triggers the signIn process.
+        console.log(credentials);
         await dbConnect();
         if (credentials?.password! || credentials?.username! === null)
-          return null;
+          throw new Error("Please fill Out all Fields");
         try {
           const user = await getOneUser(credentials?.username!);
 
-          if (user) {
+          if (!user) {
```

Comment:
```suggestion
          if (user) {
```
This is backwards. You're trying to validate the password if you DON'T have a user. You should do one step at a time and test your code multiple times. These kind of errors should be caught in testing.